cppForSwig/BlockDataMap.cpp

```diff
@@ -70,7 +79,32 @@ void BlockData::deserialize(const uint8_t* data, size_t size,
          offsetOuts[y],
          offsetOuts[y + 1] - offsetOuts[y]));
 
-      tx->lockTime_ = READ_UINT32_LE(brr.getCurrPtr() + offsetOuts.back());
+      // Add all the old serialization data to noWitData_
+      tx->noWitData_.append(brr.getCurrPtr(), 4);
+      uint32_t txInOutSize = offsetOuts.back() - offsetIns[0];
+      uint32_t viInLen;
+      if(useWitnesses) {
+         BtcUtils::readVarInt(brr.getCurrPtr() + 6, &viInLen);
+         tx->noWitData_.append(brr.getCurrPtr() + 4 + 2, txInOutSize + viInLen);
+      }
+      else {
+         BtcUtils::readVarInt(brr.getCurrPtr() + 4, &viInLen);
+         tx->noWitData_.append(brr.getCurrPtr() + 4, txInOutSize + viInLen);
```

Comment:
nowitdata should just be ignored for legacy tx. it's several orders of magnitude faster to recheck the bool flag on the fly than to copy a binary blob once